trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/SigningContext.java

```diff
@@ -15,5 +15,9 @@
 
 public interface SigningContext
 {
-    String authorization();
+    ParsedAuthorization signingAuthorization();
```

Comment:
Seems odd that this is in both the SigningContext and the Requests. wdyt?

Reply:
Though I guess it's actually a different value.

Reply:
The `ParsedAuthorization` in the request denotes the authorization information in the incoming request, while the one in the `SigningContext` is the authorization information in the newly built one. The two are not identical, but I see how "Parsed" Authorization may not be the best name.

What about `RequestAuthorization`?

Reply:
I'm tempted to move the `session` field into here too, so that it can denote all authorization information that is available in the request at the point of ingress (before we load credentials and do more stuff)